cri-dockerd: restart docker.service

krystianmlynek opened the pull request:
<!--  Thanks for sending a pull request!  Here are some tips for you:

1. If this is your first time, please read our contributor guidelines: https://git.k8s.io/community/contributors/guide/first-contribution.md and developer guide https://git.k8s.io/community/contributors/devel/development.md
2. Please label this pull request according to what type of issue you are addressing, especially if this is a release targeted pull request. For reference on required PR/issue labels, read here:
https://git.k8s.io/community/contributors/devel/sig-release/release.md#issuepr-kind-label
3. Ensure you have added or ran the appropriate tests for your PR: https://git.k8s.io/community/contributors/devel/sig-testing/testing.md
4. If you want *faster* PR reviews, read how: https://git.k8s.io/community/contributors/guide/pull-requests.md#best-practices-for-faster-reviews
5. Follow the instructions for writing a release note: https://git.k8s.io/community/contributors/guide/release-notes.md
6. If the PR is unfinished, see how to mark it: https://git.k8s.io/community/contributors/guide/pull-requests.md#marking-unfinished-pull-requests
-->

**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:
>
> /kind api-change

/kind bug
> /kind cleanup
> /kind design
> /kind documentation
> /kind failing-test
> /kind feature
> /kind flake

**What this PR does / why we need it**:
With this change containerd will be reloaded which will allow docker to pick up proxy/registry mirrors settings. Also ` cri-dockerd.socket` will be enabled to avoid kubelet failure after node reboot.

**Which issue(s) this PR fixes**:
<!--
*Automatically closes linked issue when PR is merged.
Usage: `Fixes #<issue number>`, or `Fixes (paste link of issue)`.
_If PR is about `failing-tests or flakes`, please post the related issues/tests in a comment and do not use `Fixes`_*
-->
Fixes #9142 

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:
<!--
If no, just write "NONE" in the release-note block below.
If yes, a release note is required:
Enter your extended release note in the block below. If the PR requires additional action from users switching to the new release, include the string "action required".
-->
```release-note
NONE
```

cristicalin commented on roles/container-engine/cri-dockerd/handlers/main.yml:
```diff
@@ -3,6 +3,7 @@
   command: /bin/true
   notify:
     - cri-dockerd | reload systemd
+    - reload containerd
```
Since this is part of the `cri-containerd` role the task should contain the role name like the other.

krystianmlynek replied:
Thanks. I updated the task name.

cristicalin commented:
There is another ansible-lint error: https://gitlab.com/kargo-ci/kubernetes-sigs-kubespray/builds/2879676234

krystianmlynek commented:
> There is another ansible-lint error: https://gitlab.com/kargo-ci/kubernetes-sigs-kubespray/builds/2879676234

@cristicalin Linter fixed. I used generic name task since in `tasks/main.yml` pattern of role name in task name is not followed

cristicalin commented:
Please check the reason for the docker jobs failing they, look related to the change in cri-dockerd since now we rely on cri-dockerd to support docker.

krystianmlynek commented:
@cristicalin 
I fixed the CI, but I think my assumption was wrong. I checked logs and I noticed that handlers from `container-engine/docker` are not executed - if you check this log for example https://gitlab.com/kargo-ci/kubernetes-sigs-kubespray/-/jobs/2892391348 in this part

```
TASK [container-engine/docker : Flush handlers] ********************************
[1;30mtask path: /builds/kargo-ci/kubernetes-sigs-kubespray/roles/container-engine/docker/tasks/systemd.yml:67[0m
Tuesday 16 August 2022  21:06:43 +0000 (0:00:00.118)       0:02:33.438 ******** 

RUNNING HANDLER [container-engine/docker : restart docker] *********************
[1;30mtask path: /builds/kargo-ci/kubernetes-sigs-kubespray/roles/container-engine/docker/handlers/main.yml:2[0m
Tuesday 16 August 2022  21:06:43 +0000 (0:00:00.010)       0:02:33.449 ******** 
[0;34mMETA: ran handlers[0m

TASK [container-engine/docker : ensure docker service is started and enabled] ***
[1;30mtask path: /builds/kargo-ci/kubernetes-sigs-kubespray/roles/container-engine/docker/tasks/main.yml:163[0m
Tuesday 16 August 2022  21:06:43 +0000 (0:00:00.109)       0:02:33.558 ******** 
```
it's visible that handlers(reload systemd, reload docker.socket, reload docker) were not executed hence causing my initial issue with proxy/registry mirrors settings in docker. 
I also checked logs from my deployment on bare-metal and flush_handlers in this part is not executed at all. 
I changed this flush_handler to normal execution of these task just to check and my deployment was okay - proxy and mirrors were correct. 

Current solution in this PR is working, but I think it would be better to investigate the docker part. Would it be possible for you to check this handler issue somewhere on kubespray side? 
Thanks

cristicalin commented on roles/container-engine/cri-dockerd/tasks/main.yml:
```diff
@@ -21,5 +21,13 @@
   notify:
     - restart cri-dockerd
+
+- name: Reload systemd and restart containerd   # noqa no-handler
```
Shouldn't this actually restart the `docker` service which in turn should trigger the `containerd` restart ? In this situation `cri-dockerd` talks to docker through the docker socket not to containerd directly, that is kind of the point of maintaining this compatibility layer, we should keep the same abstraction in our code.

krystianmlynek replied:
Thanks, I checked restarting of docker.service on my end and got correct results. Also switched back to using handler here with flush_handlers since on my end the previous CI error didn't happen. Hope it will be the same for CI here.

krystianmlynek commented:
/retest
